### ISSUE TITLE
Reset date range does not update the since and until date hashes

### ISSUE THREAD

**Issue by jamessspanggg** [reposense/RepoSense#1044]
To reproduce: modify the link since/until date url hashes. Reload the page. Click the reset date range button. Reload again to see that the dates are not updated. [link](https://deploy-preview-1008--reposense.netlify.com/#search=&sort=groupTitle&sortWithin=title&since=2017-10-09&timeframe=commit&mergegroup=false&groupSelect=groupByRepos&breakdown=false&until=2019-01-20)

![Kapture 2020-02-03 at 0 08 19](https://user-images.githubusercontent.com/32864116/73611105-6fada880-4619-11ea-8a96-d548b3306ee4.gif)

**Comment by Tejas2805** [comment]
It seems to be working fine on mine.

**Comment by jamessspanggg** [comment]
@Tejas2805 can you try to following the steps to see if the same bug can be reproduced?

> To reproduce: modify the link since/until date url hashes. Reload the page. Click the reset date range button. Reload again to see that the dates are not updated. [link](https://deploy-preview-1008--reposense.netlify.com/#search=&sort=groupTitle&sortWithin=title&since=2017-10-09&timeframe=commit&mergegroup=false&groupSelect=groupByRepos&breakdown=false&until=2019-01-20)

**Comment by ccyccyccy** [comment]
I believe this bug only happens you do not specify since and until date in the command line to generate the report, which as a result sets the `hasModifiedSinceDate`  and  `hasModifiedUntilDate` to false on initial loading of the page.

I can do a PR to bypass the check below, but I believe we can remove the `hasModifiedSinceDate` and `hasModifiedUntilDate` all together? I'm not too sure if they serve any purpose? @jamessspanggg 
![image](https://user-images.githubusercontent.com/32454748/73732906-fc6d7900-4775-11ea-990a-6165e21dc390.png)

**Comment by 0blivious** [comment]
The code here did serve a purpose. It stores the date range and help restore the page when refresh. Pls dont remove it.

**Comment by ccyccyccy** [comment]
Oops I mean removing the if() statements.
So instead of
```
if (this.hasModifiedSinceDate) {
        addHash('since', this.filterSinceDate);
}

if (this.hasModifiedUntilDate) {
        addHash('until', this.filterUntilDate);
}
```
We can change it to
```
addHash('since', this.filterSinceDate);
addHash('until', this.filterUntilDate);
```

**Comment by ccyccyccy** [comment]
I believe that the `hasModifiedSinceDate` indicates whether a value has been specified in the command line argument to generate the report, as written in the line 
`hasModifiedSinceDate: window.app.isSinceDateProvided`
in `v_summary.js`, but is currently interpreted as whether a filter is set by the user on the dashboard.

**Comment by 0blivious** [comment]
I feel we should not add the hash if there is no need to do so. Maybe you should change the reset date function to rectify the behavior.

**Comment by ccyccyccy** [comment]
> I feel we should not add the hash if there is no need to do so. Maybe you should change the reset date function to rectify the behavior.

Thanks for the suggestion. I implemented a simple fix (#1044) that would solve the issue but I feel that it's not very maintainable.

> I believe that the `hasModifiedSinceDate` indicates whether a value has been specified in the command line argument to generate the report, as written in the line
> `hasModifiedSinceDate: window.app.isSinceDateProvided`
> in `v_summary.js`, but is currently interpreted as whether a filter is set by the user on the dashboard.

@jamessspanggg Can you help verify this? I'm not too certain if this is the case. The code was authored by you in #757

**Comment by jamessspanggg** [comment]
> I believe that the hasModifiedSinceDate indicates whether a value has been specified in the command line argument to generate the report,

@0blivious is correct. The purpose of `hasModifiedSinceDate` and `hasModifiedUntilDate` was to indicate whether the dates are specified in the CLI. This is because prof wants a way to always display the latest since/until date, if he did not specify the since/until date in the CLI. With this, prof does not need to always send a new link whenever he updates the analysis, but students can use the same link to get the most updated since/until date.
